[2024-12-09T12:30:20.351+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-09T12:30:20.365+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparking_flow.python_job manual__2024-12-09T12:30:18.154760+00:00 [queued]>
[2024-12-09T12:30:20.370+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparking_flow.python_job manual__2024-12-09T12:30:18.154760+00:00 [queued]>
[2024-12-09T12:30:20.371+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-09T12:30:20.379+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): python_job> on 2024-12-09 12:30:18.154760+00:00
[2024-12-09T12:30:20.389+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=186) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-09T12:30:20.389+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'sparking_flow', 'python_job', 'manual__2024-12-09T12:30:18.154760+00:00', '--job-id', '412', '--raw', '--subdir', 'DAGS_FOLDER/spark_***.py', '--cfg-path', '/tmp/tmp0n138kzr']
[2024-12-09T12:30:20.394+0000] {standard_task_runner.py:72} INFO - Started process 188 to run task
[2024-12-09T12:30:20.394+0000] {standard_task_runner.py:105} INFO - Job 412: Subtask python_job
[2024-12-09T12:30:20.452+0000] {task_command.py:467} INFO - Running <TaskInstance: sparking_flow.python_job manual__2024-12-09T12:30:18.154760+00:00 [running]> on host 2a892a3469ec
[2024-12-09T12:30:20.525+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Yusuf Ganiyu' AIRFLOW_CTX_DAG_ID='sparking_flow' AIRFLOW_CTX_TASK_ID='python_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-09T12:30:18.154760+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-09T12:30:18.154760+00:00'
[2024-12-09T12:30:20.526+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-09T12:30:20.549+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2024-12-09T12:30:20.551+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py
[2024-12-09T12:30:26.377+0000] {spark_submit.py:634} INFO - Downloading to: /tmp/test1234.csv
[2024-12-09T12:30:26.388+0000] {spark_submit.py:634} INFO - File downloaded to /tmp/test1234.csv
[2024-12-09T12:30:26.640+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:26 INFO SparkContext: Running Spark version 3.5.3
[2024-12-09T12:30:26.648+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:26 INFO SparkContext: OS info Linux, 6.10.14-linuxkit, aarch64
[2024-12-09T12:30:26.649+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:26 INFO SparkContext: Java version 17.0.13
[2024-12-09T12:30:26.796+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:26 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-09T12:30:27.041+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO ResourceUtils: ==============================================================
[2024-12-09T12:30:27.042+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-12-09T12:30:27.043+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO ResourceUtils: ==============================================================
[2024-12-09T12:30:27.044+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO SparkContext: Submitted application: CSVReadExample
[2024-12-09T12:30:27.119+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-12-09T12:30:27.140+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO ResourceProfile: Limiting resource is cpu
[2024-12-09T12:30:27.145+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-12-09T12:30:27.264+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO SecurityManager: Changing view acls to: ***
[2024-12-09T12:30:27.265+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO SecurityManager: Changing modify acls to: ***
[2024-12-09T12:30:27.268+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO SecurityManager: Changing view acls groups to:
[2024-12-09T12:30:27.269+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO SecurityManager: Changing modify acls groups to:
[2024-12-09T12:30:27.273+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-12-09T12:30:27.916+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO Utils: Successfully started service 'sparkDriver' on port 43441.
[2024-12-09T12:30:27.963+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:27 INFO SparkEnv: Registering MapOutputTracker
[2024-12-09T12:30:28.025+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO SparkEnv: Registering BlockManagerMaster
[2024-12-09T12:30:28.052+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-12-09T12:30:28.056+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-12-09T12:30:28.072+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-12-09T12:30:28.161+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-1a9b1bdd-5928-4545-88b9-a39c2c6c44fe
[2024-12-09T12:30:28.211+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-12-09T12:30:28.255+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-12-09T12:30:28.707+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-12-09T12:30:28.894+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-12-09T12:30:28.895+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.
[2024-12-09T12:30:28.934+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:28 INFO Utils: Successfully started service 'SparkUI' on port 4042.
[2024-12-09T12:30:29.582+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:29 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://sparkingflow-spark-master-1:7077...
[2024-12-09T12:30:29.813+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:29 INFO TransportClientFactory: Successfully created connection to sparkingflow-spark-master-1/172.18.0.5:7077 after 113 ms (0 ms spent in bootstraps)
[2024-12-09T12:30:30.170+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241209123030-0001
[2024-12-09T12:30:30.196+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 46881.
[2024-12-09T12:30:30.197+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO NettyBlockTransferService: Server created on 2a892a3469ec:46881
[2024-12-09T12:30:30.200+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-12-09T12:30:30.223+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2a892a3469ec, 46881, None)
[2024-12-09T12:30:30.231+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO BlockManagerMasterEndpoint: Registering block manager 2a892a3469ec:46881 with 434.4 MiB RAM, BlockManagerId(driver, 2a892a3469ec, 46881, None)
[2024-12-09T12:30:30.235+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2a892a3469ec, 46881, None)
[2024-12-09T12:30:30.239+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2a892a3469ec, 46881, None)
[2024-12-09T12:30:30.795+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:30 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-12-09T12:30:31.254+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:31 INFO SparkContext: Added file /tmp/test1234.csv at spark://2a892a3469ec:43441/files/test1234.csv with timestamp 1733747431253
[2024-12-09T12:30:31.256+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:31 INFO Utils: Copying /tmp/test1234.csv to /tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv
[2024-12-09T12:30:31.285+0000] {spark_submit.py:634} INFO - Resolved file path by Spark: /tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv
[2024-12-09T12:30:31.286+0000] {spark_submit.py:634} INFO - file_path /tmp/test1234.csv
[2024-12-09T12:30:31.301+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:31 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-12-09T12:30:31.305+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:31 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-12-09T12:30:33.138+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 WARN FileSystem: Failed to initialize fileystem hdfs:///tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv: java.io.IOException: Incomplete HDFS URI, no host: hdfs:///tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv
[2024-12-09T12:30:33.145+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: hdfs:////tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv.
[2024-12-09T12:30:33.146+0000] {spark_submit.py:634} INFO - java.io.IOException: Incomplete HDFS URI, no host: hdfs:///tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv
[2024-12-09T12:30:33.146+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:184)
[2024-12-09T12:30:33.146+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3469)
[2024-12-09T12:30:33.147+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
[2024-12-09T12:30:33.147+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
[2024-12-09T12:30:33.147+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
[2024-12-09T12:30:33.148+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
[2024-12-09T12:30:33.148+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
[2024-12-09T12:30:33.149+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:53)
[2024-12-09T12:30:33.149+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:366)
[2024-12-09T12:30:33.149+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-12-09T12:30:33.150+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-12-09T12:30:33.150+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-12-09T12:30:33.151+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-12-09T12:30:33.151+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:186)
[2024-12-09T12:30:33.151+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-12-09T12:30:33.152+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-12-09T12:30:33.152+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-12-09T12:30:33.153+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-12-09T12:30:33.154+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-12-09T12:30:33.155+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-09T12:30:33.156+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-12-09T12:30:33.158+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-12-09T12:30:33.161+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-12-09T12:30:33.161+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-09T12:30:33.162+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-09T12:30:33.162+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-09T12:30:33.162+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 WARN FileSystem: Failed to initialize fileystem hdfs:///tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv: java.io.IOException: Incomplete HDFS URI, no host: hdfs:///tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv
[2024-12-09T12:30:33.202+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-12-09T12:30:33.202+0000] {spark_submit.py:634} INFO - File "/opt/***/jobs/python/wordcountjob.py", line 42, in <module>
[2024-12-09T12:30:33.203+0000] {spark_submit.py:634} INFO - df = spark.read.format("csv").option("header","false").load('hdfs:///'+myfile)
[2024-12-09T12:30:33.204+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-09T12:30:33.205+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 307, in load
[2024-12-09T12:30:33.206+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2024-12-09T12:30:33.206+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2024-12-09T12:30:33.207+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2024-12-09T12:30:33.208+0000] {spark_submit.py:634} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o27.load.
[2024-12-09T12:30:33.209+0000] {spark_submit.py:634} INFO - : java.io.IOException: Incomplete HDFS URI, no host: hdfs:///tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/userFiles-850f5431-95fa-4c68-8370-0aec84c7a6d1/test1234.csv
[2024-12-09T12:30:33.210+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:184)
[2024-12-09T12:30:33.210+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3469)
[2024-12-09T12:30:33.211+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
[2024-12-09T12:30:33.212+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
[2024-12-09T12:30:33.212+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
[2024-12-09T12:30:33.212+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
[2024-12-09T12:30:33.213+0000] {spark_submit.py:634} INFO - at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
[2024-12-09T12:30:33.214+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$1(DataSource.scala:724)
[2024-12-09T12:30:33.214+0000] {spark_submit.py:634} INFO - at scala.collection.immutable.List.map(List.scala:293)
[2024-12-09T12:30:33.215+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.checkAndGlobPathIfNecessary(DataSource.scala:722)
[2024-12-09T12:30:33.215+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.checkAndGlobPathIfNecessary(DataSource.scala:551)
[2024-12-09T12:30:33.216+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:404)
[2024-12-09T12:30:33.217+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-12-09T12:30:33.218+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-12-09T12:30:33.218+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-12-09T12:30:33.219+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-12-09T12:30:33.220+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:186)
[2024-12-09T12:30:33.221+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-12-09T12:30:33.221+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-12-09T12:30:33.222+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-12-09T12:30:33.223+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-12-09T12:30:33.223+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-12-09T12:30:33.224+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-09T12:30:33.224+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-12-09T12:30:33.225+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-12-09T12:30:33.225+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-12-09T12:30:33.226+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-09T12:30:33.226+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-09T12:30:33.227+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-09T12:30:33.227+0000] {spark_submit.py:634} INFO - 
[2024-12-09T12:30:33.316+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO SparkContext: Invoking stop() from shutdown hook
[2024-12-09T12:30:33.317+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-12-09T12:30:33.337+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO SparkUI: Stopped Spark web UI at http://2a892a3469ec:4042
[2024-12-09T12:30:33.343+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-12-09T12:30:33.348+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-12-09T12:30:33.399+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-12-09T12:30:33.438+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO MemoryStore: MemoryStore cleared
[2024-12-09T12:30:33.439+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO BlockManager: BlockManager stopped
[2024-12-09T12:30:33.456+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-12-09T12:30:33.459+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-12-09T12:30:33.481+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO SparkContext: Successfully stopped SparkContext
[2024-12-09T12:30:33.486+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO ShutdownHookManager: Shutdown hook called
[2024-12-09T12:30:33.489+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO ShutdownHookManager: Deleting directory /tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074/pyspark-1d69d8e9-2384-42e5-b0e6-8195fda4bbe1
[2024-12-09T12:30:33.495+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO ShutdownHookManager: Deleting directory /tmp/spark-7eebeeee-3125-40e4-805b-bd25f823cf79
[2024-12-09T12:30:33.505+0000] {spark_submit.py:634} INFO - 24/12/09 12:30:33 INFO ShutdownHookManager: Deleting directory /tmp/spark-28e17f5f-09df-4e2d-a2db-bb2efc5fe074
[2024-12-09T12:30:33.854+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-09T12:30:33.888+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=sparking_flow, task_id=python_job, run_id=manual__2024-12-09T12:30:18.154760+00:00, execution_date=20241209T123018, start_date=20241209T123020, end_date=20241209T123033
[2024-12-09T12:30:33.933+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-09T12:30:33.936+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 412 for task python_job (Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.; 188)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-09T12:30:34.037+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-09T12:30:34.245+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-09T12:30:34.261+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
